src/Symfony/Component/Intl/Currencies.php

```diff
@@ -139,6 +139,156 @@ public static function forNumericCode(int $numericCode): array
+     */
+    public static function existsInCountry(
+        string $country,
+        string $currency,
+        bool $activeOnly = true,
+        string $date = 'today',
+    ): bool {
+        try {
+            $currencyMetadata = self::readEntry(['Map', $country, $currency], 'meta');
+        } catch (MissingResourceException) {
+            return false;
+        }
+
+        if (!$activeOnly) {
+            return true;
+        }
+
+        return self::isActive($country, $currency, $currencyMetadata, $date);
+    }
+
+    /**
+     * @param array{tender?: bool} $currencyMetadata
+     */
+    private static function isLegalTender(array $currencyMetadata): bool
+    {
+        return !\array_key_exists('tender', $currencyMetadata) || false !== $currencyMetadata['tender'];
+    }
+
+    /**
+     * @param non-empty-string $country  e.g. 'FR'
+     * @param non-empty-string $currency  e.g. 'USD'
+     * @param array{from?: non-empty-string, to?: non-empty-string} $currencyMetadata
+     * @param string $date The date on which the check will be performed
+     */
+    private static function isActive(string $country, string $currency, array $currencyMetadata, string $date = 'today'): bool
+    {
+        if (!\array_key_exists('from', $currencyMetadata)) {
+            // Note: currencies that are not legal tender don't have often validity dates.
+            throw new \RuntimeException("Cannot check whether the currency $currency is active or not because they are no validity dates available.");
+        }
+
+        $from = \DateTimeImmutable::createFromFormat('Y-m-d', $currencyMetadata['from']);
+
+        if (!$from) {
+            throw new \RuntimeException("Unable to parse the `from` date for currency $currency in country $country.");
+        }
+
+        if (\array_key_exists('to', $currencyMetadata)) {
+            $to = \DateTimeImmutable::createFromFormat('Y-m-d', $currencyMetadata['to']);
+
+            if (!$to) {
+                throw new \RuntimeException("Unable to parse the `to` date for currency $currency in country $country.");
+            }
+        } else {
+            $to = null;
+        }
+
+        try {
+            $date = new \DateTimeImmutable($date);
+        } catch (\DateMalformedStringException $exception) {
+            throw new \InvalidArgumentException("Invalid date provided: $date.", previous: $exception);
+        }
+
+        return $from <= $date && (null === $to || $to >= $date);
+    }
+
+    /**
+     * @param non-empty-string $currency e.g. 'USD'
+     * @param ?bool $legalTender if the currency must be a legal tender. Using null do not filter anything
+     * @param bool $activeOnly exclude currencies that are not active on the given $date in countries
+     * @param string $date the date on which the check will be performed if $activeOnly is set to true
+     */
+    public static function existsInAtLeastOneCountry(
```

Comment:
this feels a bit weird naming compared to `exists()`, which should also checks existence in at least 1 country